app/components/shared/main-navigation.component.ts

```diff
@@ -1,8 +1,24 @@
 import { Component } from "@angular/core";
 
+import { AccountService } from "app/services";
+
 @Component({
     selector: "bl-app-nav",
     templateUrl: "main-navigation.html",
 })
 export class MainNavigationComponent {
+
+    public selectedId: string;
+    public selectedAccountAlias: string = "";
+
+    constructor(private accountService: AccountService) {
+        accountService.currentAccountId.subscribe((accountId) => {
+            if (accountId) {
+                this.selectedId = accountId;
+                this.selectedAccountAlias = accountService.getNameFromAccountId(accountId);
+            } else {
+                this.selectedAccountAlias = "No account selected!";
```

Comment:
Alias is used as the display name. AccountId is the full Resource Id e.g. /subscription/.../resourceGroup/.../batchAccount/foo